From bcd0577ad7d5f38f2072a74831bf596eb306980c Mon Sep 17 00:00:00 2001
From: sdinkinx <simonx.dinkin@intel.com>
Date: Sun, 24 Jun 2018 15:10:45 +0300
Subject: [PATCH] nl80211: change sta rssi to be optional in
 nl80211_new_station

don't return error in case sta rssi was not set in
nl80211_new_station function.

Signed-off-by: sdinkinx <simonx.dinkin@intel.com>
---
 net/wireless/nl80211.c | 6 ++----
 1 file changed, 2 insertions(+), 4 deletions(-)

diff --git a/net/wireless/nl80211.c b/net/wireless/nl80211.c
index 3aa1943..4271f5a 100644
--- a/net/wireless/nl80211.c
+++ b/net/wireless/nl80211.c
@@ -4058,8 +4058,8 @@ static int nl80211_new_station(struct sk_buff *skb, struct genl_info *info)
 	if (!info->attrs[NL80211_ATTR_STA_AID])
 		return -EINVAL;
 
-	if (!info->attrs[NL80211_ATTR_STA_RSSI])
-		return -EINVAL;
+	if (info->attrs[NL80211_ATTR_STA_RSSI])
+		params.rssi = nla_get_u32(info->attrs[NL80211_ATTR_STA_RSSI]);
 
 	mac_addr = nla_data(info->attrs[NL80211_ATTR_MAC]);
 	params.supported_rates =
@@ -4078,8 +4078,6 @@ static int nl80211_new_station(struct sk_buff *skb, struct genl_info *info)
 	if (!params.aid || params.aid > IEEE80211_MAX_AID)
 		return -EINVAL;
 
-	params.rssi = nla_get_u32(info->attrs[NL80211_ATTR_STA_RSSI]);
-
 	if (info->attrs[NL80211_ATTR_STA_CAPABILITY]) {
 		params.capability =
 			nla_get_u16(info->attrs[NL80211_ATTR_STA_CAPABILITY]);
-- 
2.10.1

